client/request_test.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+)
+
+// TestSetHostHeader should set fake host for local communications, set real host
+// for normal communications.
+func TestSetHostHeader(t *testing.T) {
+	testURL := "/test"
+	testCases := []struct {
+		host            string
+		expectedHost    string
+		expectedURLHost string
+	}{
+		{
+			"unix:///var/run/docker.sock",
+			"",
+			"docker",
+		},
+		{
+			"npipe:////./pipe/docker_engine",
+			"",
+			"docker",
+		},
+		{
+			"tcp://0.0.0.0:4243",
+			"",
```

Comment:
oh yep, I think it would be better to use `NewClient` everywhere.

- renaming `newMockClient` to `newMockTransportClient` (or `newMockTransport`…) as it's more what it does.
- probably creating a `newMockClient` that does call `NewClient`, …

```go
func newMockClient(tlsConfig *tls.Config, doer func(*http.Request) (*http.Response, error)) Client {
        client, _ := NewClient(test.host, "", nil, nil)

        client.transport = newMockTransport(tlsConfig, doer)

        return client
}
```

I would even think we should check the error on `NewClient`.